src/bgw/job_stat.c

```diff
@@ -203,9 +203,9 @@ calculate_next_start_on_failure(TimestampTz finish_time, int consecutive_failure
 	float8 multiplier = (consecutive_failures > MAX_FAILURES_MULTIPLIER ? MAX_FAILURES_MULTIPLIER :
 																		  consecutive_failures);
 	MemoryContext oldctx;
-	Assert(consecutive_failures > 0);
-	int64 max_slots =
-		(1 << consecutive_failures) - 1; /* 2^(consecutive_failures) - 1, at most 2^20 */
+	Assert(consecutive_failures > 0 && multiplier < 63);
+	/* 2^(consecutive_failures) - 1, at most 2^20 */
+	int64 max_slots = (INT64CONST(1) << (int64) multiplier) - INT64CONST(1);
```

Comment:
Not a huge fan of `(int64)(float8)(int)` but approving.

Reply:
Yeah, why are we using floats there?

Reply:
for the call 
```
ival = DirectFunctionCall2(interval_mul, IntervalPGetDatum(&job->fd.retry_period), Float8GetDatum(multiplier));
```
on line 237

Reply:
Ah!